File: LeetCode_List/206Reverse_Linked_List(final).cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
// 还是觉得这份代码最叼 没有之一 毕竟是采用了递归的思想

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if (head==NULL || head->next==NULL) { return head; }
     	ListNode *p=reverseList(head->next);
     	head->next->next=head;
     	head->next=NULL;
     	return p;
    }
};

File: LeetCode_List/206Reverse_Linked_List(final).py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

# 啊哈哈哈 牛逼如我 想了一下那个回文链表的list怎么写 就一次通过了！

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        rev=None
    	while head:
    		rev,rev.next,head=head,rev,head.next
    	return rev